src/py42/services/users.py

```diff
@@ -47,7 +48,12 @@ def create_user(
             u"lastName": last_name,
             u"notes": notes,
         }
-        return self._connection.post(uri, json=data)
+
+        response = self._connection.post(uri, json=data)
+        if response.status_code == 500 and response.text == "USER_DUPLICATE":
```

Comment:
This will raise a `Py42InternalServerError` before reaching line 53. So we should just checking the status code with a `try/catch` for `Py42InternalServerError` and _then_ check the response text, which is available from the exception itself via `error.response.text`

Reply:
Thanks for the information here! I feel this PR should be good to go for ya.

Sorry for the style churn, but it was fun contributing to this :)